### DISCUSSION TITLE
How can I backport 0.38's ExtrinsicDetails::hash to 0.37?

### DISCUSSION THREAD

**Discussion by jmg-duarte** [paritytech/subxt#1851] [Q&A]
I've tried to do the following:
```
let extrinsic_hash = <ParachainConfig as subxt::Config>::Hasher::hash(extrinsic.bytes());
```
But it does not match the hash returned by `sign_and_submit_default`, even though the `bytes()` method shouldn't really change?

If I use version 0.38 all this works, I can't upgrade since I was facing issues with Zombienet's SDK which Cargo was making pick up 0.37 (https://github.com/paritytech/zombienet-sdk/pull/271)

**Comment by niklasad1** [comment]
Hey @jmg-duarte 

Can you explain in more detail what the issue is? Does it return a different hash from the same extrinsic bytes or just that it doesn't compile because of different types?

IIRC, we bumped primitives_types to v0.13.1 in subxt which most likely will cause compilation issues if the `type Hash = H256` in the subxt::Config.

**Reply by jmg-duarte** [reply]
It returns a different hash. To work around https://github.com/paritytech/subxt/issues/1668 I'm subscribing to the finalized block stream and looking for the extrinsic on another task. So the code amounts to something like:

```
let xt_hash = client.sign_and_submit_default().await?;
...
let extrinsic_hash = <ParachainConfig as subxt::Config>::Hasher::hash(extrinsic.bytes());

if xt_hash == extrinsic_hash {
    // never happens
}
```

Regardless, I found what I needed:
```
for extrinsic in block.extrinsics().await?.iter() {
                    if extrinsic.events().await?.extrinsic_hash() == extrinsic_hash {
```
However, this is broken in 0.38... the check always fails, never finding the right hash.

Thanks for the help!
